torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -84,6 +84,9 @@ std::vector<int64_t> ProcessGroupAgent::MessageCounter::snapshot() {
 
 ////////////////////////  ProcessGroupAgent  /////////////////////////////////
 
+constexpr std::chrono::milliseconds INFINITY_SLEEP_INTERVAL =
+    std::chrono::hours(8760);
```

Comment:
How do we get 8760? Can we do this instead `std::chrono::duration<int64_t, std::nano>(std::numeric_limits<int64_t>::max())`

Reply:
I don't think this works with `std::chrono::milliseconds` (at least this particular line). How about `const INTERVAL = std::chrono::seconds(std::numeric_limits<int64_t>::max()` or something of that nature?

Reply:
I think you will need to use std::chrono::nanoseconds since that is the only one that supports 64 bit value (I don't think int64_t max would work with seconds). See `Helper Types` here: https://en.cppreference.com/w/cpp/chrono/duration.